nixos/modules/programs/hyprland.nix

```diff
@@ -0,0 +1,87 @@
+{ config
+, lib
+, pkgs
+, ...
+}:
+with lib; let
+  cfg = config.programs.hyprland;
+
+  defaultHyprlandPackage = pkgs.hyprland.override {
+    enableXWayland = cfg.xwayland.enable;
+    hidpiXWayland = cfg.xwayland.hidpi;
+    nvidiaPatches = cfg.nvidiaPatches;
+  };
+in
+{
+  options.programs.hyprland = {
+    enable = mkEnableOption null // {
+      description = mdDoc ''
+        Hyprland, the dynamic tiling Wayland compositor that doesn't sacrifice on its looks.
+
+        You can manually launch Hyprland by executing "exec Hyprland" on a TTY.
+
+        A configuration file will be generated in ~/.config/hypr/hyprland.conf.
+        See <https://wiki.hyprland.org> for more information.
+      '';
+    };
+
+    package = mkOption {
+      type = types.path;
+      default = defaultHyprlandPackage;
+      defaultText = literalExpression ''
+        pkgs.hyprland.override {
+          enableXWayland = cfg.xwayland.enable;
+          hidpiXWayland = cfg.xwayland.hidpi;
+          nvidiaPatches = cfg.nvidiaPatches;
+        }
+      '';
+      example = literalExpression "<Hyprland flake>.packages.<system>.default";
+      description = mdDoc ''
+        The Hyprland package to use.
+        Setting this option will make {option}`programs.hyprland.xwayland` and
+        {option}`programs.hyprland.nvidiaPatches` not work.
+      '';
+    };
+
+    xwayland = {
+      enable = mkEnableOption (mdDoc "XWayland") // { default = true; };
+      hidpi = mkEnableOption null // {
+        description = mdDoc ''
+          Enable HiDPI XWayland, based on [XWayland MR 733](https://gitlab.freedesktop.org/xorg/xserver/-/merge_requests/733).
+          See <https://wiki.hyprland.org/Nix/Options-Overrides/#xwayland-hidpi> for more info.
+        '';
+      };
+    };
+
+    nvidiaPatches = mkEnableOption (mdDoc "patching wlroots for better Nvidia support");
+  };
+
+  config = mkIf cfg.enable {
+    environment = {
+      systemPackages = [ cfg.package ];
+
+      sessionVariables = {
+        NIXOS_OZONE_WL = mkDefault "1";
+      };
```

Comment:
I'm a little hesitant to force this by default even though it's not as experimental of an environment variable anymore

Does anyone else have thoughts on this?

(And perhaps at some point the env var should just be removed and wayland support automatically be enabled at all chromium-ish packages -- who knows if/when that'll happen though)

Reply:
I haven't heard anyone complain about it in Hyprland's issues, nor in the Discord server. I've used it since the beginning, and I haven't had issues.

Reply:
The concern is less that it may cause issues but that it's an experimental flag which is still expected to be disabled by default, and currently no NixOS or home-manager modules set it as a session var across all sessions like this

But maybe whether it actually causes issues should be the bigger concern than it's now year-and-a-half experimental status idk, since realistically it gives a significantly better user experience in general when running under a Wayland compositor

I think someone more authoritative should probably weigh in, maybe @wmertens who added the env var initially?

Reply:
Having this flag for example breaks signal-desktop. We currently set it on a package level instead of having it globally enabled.

Reply:
https://github.com/NixOS/nixpkgs/pull/222163

Reply:
I'm still skeptical of enabling the env var by default, but wouldn't that PR (#222163) mean setting it no longer breaks signal-desktop? (Or did I misunderstand what you meant?)

Reply:
Ah, indeed. you are right.

Reply:
Either way, I do agree we just leaving it off by default but possibly suggesting it (or providing an option for it?) is the more prudent move here